make/Images.gmk

```diff
@@ -122,32 +128,35 @@ define CreateCDSArchive
   $1_$2_DUMP_EXTRA_ARG := $(if $(filter _nocoops, $2),-XX:-UseCompressedOops,)
   $1_$2_DUMP_TYPE      := $(if $(filter _nocoops, $2),-NOCOOPS,)
 
+  # Only G1 supports dumping the shared heap, so explicitly use G1 if the JVM supports it.
+  $1_$2_CDS_DUMP_FLAGS := $(CDS_DUMP_FLAGS) $(if $(filter g1gc, $(JVM_FEATURES_$1)),-XX:+UseG1GC)
+
   ifeq ($(OPENJDK_TARGET_OS), windows)
     $1_$2_CDS_ARCHIVE := bin/$1/classes$2.jsa
   else
     $1_$2_CDS_ARCHIVE := lib/$1/classes$2.jsa
   endif
 
   $$(eval $$(call SetupExecute, $1_$2_gen_cds_archive_jdk, \
-      WARN := Creating CDS$$($1_$2_DUMP_TYPE) archive for jdk image for $1, \
+      WARN := Creating CDS$$($1_$2_DUMP_TYPE) archive for jdk image for $1: $$($1_$2_CDS_DUMP_FLAGS), \
```

Comment:
The actual flags is too much detail to put on the highest (badly named) WARN level. Instead introduce an additional INFO output, something like this:
```
 INFO := Using CDS flags for $1: $$($1_$2_CDS_DUMP_FLAGS), \
```